nomad-cluster-security.md

```diff
@@ -0,0 +1,14 @@
+# Nomad Cluster Security
+
+1. What is the noamd cluster size?
+1. Is Nomad deployed single region or multi-region? What is the deployment strategy? How is the high availability and failover designed? How is leader and follower selected?
+1. Is Nomad integrated with consul? How the DNS resolutions works to connect to the consul cluster?
+1. How is Nomad to Consul connectivity? Is it over HTTP secured with TLS? And is there any consul token to encrypt the traffic?
+1. Is the replication happens between the Nomad cluster in different regions?
+1. Does Nomad cluster uses the ACL policies, Tokens and sentinal policies?
+1. Do we have bootstrap Nomad ACL systems? it will make sure we have tokens , policies - rules and capabilities
+1. Are you using vault for secret management? Is your vault has the version support?
+1. How the services are isolated in Nomad cluster?
+1. How the database connectivity works in Nomad cluster? How about the permission on persistent storage mounted?
```

Comment:
Please clarify these questions, especially also the second. What kind of answer are you after?

Reply:
Tokens - Requests to Nomad are authenticated using a bearer token. Each ACL token has a public Accessor ID which is used to name a token and a Secret ID which is used to make requests to Nomad. So , all the jobs submitted need to talk to the Nomad API's.

Policies - Policies consist of a set of rules defining the capabilities or actions to be granted. For example, a "readonly" policy might only grant the ability to list and inspect running jobs, but not to submit new ones

Capabilities - Capabilities are the set of actions that can be performed. This includes listing jobs, submitting jobs, querying nodes, etc. A management token is granted all capabilities, while client tokens are granted specific capabilities via ACL policies

So as an audit , we want to check what type of tokens are generated and with what policies and capabilities .

https://learn.hashicorp.com/tutorials/nomad/access-control-bootstrap?in=nomad/access-control

Regarding -  Secrets - HashiCorp Vault has a secrets engine for generating short-lived Nomad tokens. As Vault has a number of authentication backends, it could provide a workflow where a user or orchestration system authenticates using an pre-existing identity service (LDAP, Okta, Amazon IAM, etc.) in order to obtain a short-lived Nomad token.

We want to know if they use the Vault for generating the Nomad tokens ? And If yes , Is there any authentication mechanism already present like LDAP etc .?

https://learn.hashicorp.com/tutorials/nomad/vault-nomad-secrets?in=nomad/access-control
https://www.nomadproject.io/docs/integrations/vault-integration